js/virwox.js

```diff
@@ -118,11 +118,8 @@ module.exports = class virwox extends Exchange {
             const currencyId = this.safeString (balance, 'currency');
             const code = this.commonCurrencyCode (currencyId);
             const total = this.safeFloat (balance, 'balance');
-            const account = {
-                'free': total,
```

Comment:
What should we do in this situtation where the total value is being used for both the 'free' key and the 'total' key?

Reply:
Same rule:

- if `total` is undefined, but `free` and `used` are defined – the `total` can be defined then
- if `free` in undefined, but `total` and `used` are defined – the `free` can be defined then
- if `used` in undefined, but `total` and `free` are defined – the `used` can be defined then
- if two of the values are undefined, the third value cannot be changed
- if all values are defined, neither of them can be changed (even if it yields an arithmetically incorrect formula total (0) = used (0) + free (5), but i don't think it will)

Reply:
(The above assumes that you have not edited the derived exchanges and we're talking of the base edits only – the derived exchanges we will fix one by one after merging the base edits).

Reply:
Along with merging the edits to the base class we will have to upgrade our `fetchBalance` test: https://github.com/ccxt/ccxt/blob/master/js/test/Exchange/test.fetchBalance.js

Reply:
https://github.com/ccxt/ccxt/pull/5315/commits/0d7d037eb565218ca7b3044856a7525b9ed8d8ad

What you explained is already implemented in the base edits. I'm talking more of the case where one field is being used for both free and total in the current codebase like so:

```
balance = this.safeFloat (response, 'balance');
account = {
    'free': balance,
    'used': 0.0,
    'total': balance,
}
```

even though you are assuming that you have 0.0 used here

Reply:
We should not change the base class for this, it has to be addressed in the derived exchange, I believe. 

Here's the plan:

1. The base class should not change the value of `used` in such cases – it should keep the existing value.
2. The base `fetchBalance` test should check for the arithmetics of `total = free + used` and should make sure that if two of the values are defined – the third value should be defined as well.
3. Then we open the exchange that has `used`: 0.0, change it to undefined and see if it breaks the `fetchBalance` test – for every exchange in question.